src/bun.js/api/ffi.zig

```diff
@@ -470,54 +584,107 @@ pub const FFI = struct {
         }
 
         if (object.get(globalThis, "library")) |library_value| {
-            if (library_value.isArray()) {
-                var iter = library_value.arrayIterator(globalThis);
-                compile_c.libraries.ensureTotalCapacityPrecise(bun.default_allocator, iter.len) catch bun.outOfMemory();
+            compile_c.libraries = StringArray.fromJS(globalThis, library_value, "library");
```

Comment:
this will coerce to a string with `getZigString` if the value isn't an array. do we want to error on non string values like `flags`?